core/src/main/java/io/questdb/cutlass/pgwire/modern/PGPipelineEntry.java

```diff
@@ -372,10 +376,12 @@ public void compileNewSQL(
         try {
             sqlExecutionContext.setCacheHit(cacheHit = false);
             try (SqlCompiler compiler = engine.getSqlCompiler()) {
-                // Define the provided PostgresSQL types on the BindVariableService. The compilation
-                // below will use these types to build the plan, and it will also define any missing bind
-                // variables.
-                msgParseDefineBindVariableTypes(sqlExecutionContext.getBindVariableService());
+                if (!recompile) {
```

Comment:
Please explain why we don't need to define binding variables when recompiling. 

As a reader I can only assume it's because recompilations only happen after execution failure and we always define binding var. before compilations.

Reply:
added comment in https://github.com/questdb/questdb/pull/5377/commits/f7861d6fe783a696d7b819712e2f9a0ebb6faa9d